Feature/mc 9440 create a forwarding rule (#68)
cloudops/cloudmc-api-docs: source/includes/gcp/_forwarding_rules.md

#### Forwarding rules

A forwarding rule and its corresponding IP address represent the frontend configuration of a load balancer.

> <!-------------------- LIST FORWARDING RULE -------------------->

Reviewer: RULES

Author: done

> <!-------------------- LIST FORWARDING RULE -------------------->

Reviewer 2: ```suggestion
<!-------------------- LIST FORWARDING RULES -------------------->
```

Author: done

      "creationTimestamp": "2019-10-22T15:23:35.006-07:00",
      "description": "",
      "IPAddress": "35.227.251.25",
      "IPProtocol": "TCP",
      "ipVersion": "IPV4",
      "kind": "compute#forwardingRule",
      "loadBalancingScheme": "EXTERNAL",
      "networkTier": "PREMIUM",
      "portRange": "80-80",
      "selfLink": "https://www.googleapis.com/compute/v1/projects/test-area/global/forwardingRules/forwardingrulename",
      "target": "https://www.googleapis.com/compute/v1/projects/test-area/global/targetHttpProxies/targetproxyname",
      "shortTarget": "targetproxyname",
      "id": "4724455712741277576",
      "name": "gfr-user-name"
    }
  ],
  "metadata": {
    "recordCount": 1
  }
}
```

<code>
   GET /service/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/forwardingrules
</code>

Retrieve a list of all forwarding rules

Attributes | &nbsp;
------- | -----------
`creationTimestamp`<br/>*string* | Creation timestamp in RFC3339 text format.
`description` <br/>*string* | An optional description of this resource.
`IPAddress`<br/>*string* | The IP address that this forwarding rule serves. 

> `IPProtocol`<br/>*enum* | The IP protocol to which this rule applies. The default protocol is TCP.

Reviewer: change enum to string, this is a JSON API, enum only makes sense in the backend.

Author: done

`ipVersion`<br/>*enum* | The IP version that will be used by this forwarding rule. The default IP version is IPV4.
`kind`<br/>*string* | Type of the resource.

> `loadBalancingScheme`<br/>*enum* | Specifies the forwarding rule type. Possible values are INTERNAL and EXTERNAL.

Reviewer: sinc we're only doing EXTERNAL load balancer at the moment, not sure we should mention INTERNAL here.

Author: change it to "The default value is EXTERNAL"

> `loadBalancingScheme`<br/>*enum* | Specifies the forwarding rule type. Possible values are INTERNAL and EXTERNAL.

Reviewer 2: ```suggestion
`loadBalancingScheme`<br/>*enum* | Specifies the forwarding rule type. EXTERNAL is the only value supported.
```

Author: change it to "The default value is EXTERNAL"

      "networkTier": "PREMIUM",
      "portRange": "80-80",
      "selfLink": "https://www.googleapis.com/compute/v1/projects/test-area/global/forwardingRules/forwardingrulename",
      "target": "https://www.googleapis.com/compute/v1/projects/test-area/global/targetHttpProxies/targetproxyname",
      "shortTarget": "targetproxyname",
      "id": "4724455712741277576",
      "name": "gfr-user-name"
   }
}
```

<code>GET /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/forwardingrules/:id</code>

Retrieve a forwarding rule in a given [environment](#administration-environments)

Attributes | &nbsp;
------- | -----------
`creationTimestamp`<br/>*string* | Creation timestamp in RFC3339 text format.
`description` <br/>*string* | An optional description of this resource.
`IPAddress`<br/>*string* | The IP address that this forwarding rule serves. 
`IPProtocol`<br/>*enum* | The IP protocol to which this rule applies. The default protocol is TCP.
`ipVersion`<br/>*enum* | The IP version that will be used by this forwarding rule. The default IP version is IPV4.
`kind`<br/>*string* | Type of the resource.

> `loadBalancingScheme`<br/>*enum* | Specifies the forwarding rule type. Possible values are INTERNAL and EXTERNAL.

Reviewer: see comment above.

Author: same as above

> `loadBalancingScheme`<br/>*enum* | Specifies the forwarding rule type. Possible values are INTERNAL and EXTERNAL.

Reviewer 2: ```suggestion
`loadBalancingScheme`<br/>*enum* | Specifies the forwarding rule type. EXTERNAL is the only value supported.
```

Author: same as above

`name`<br/>*string* | Name of the resource. Provided by the client when the resource is created.
`networkTier`<br/>*enum* | This signifies the networking tier used for configuring this load balancer.
`portRange`<br/>*string* | The destination ports that this forwarding rule forwards packets with.
`selfLink`<br/>*string* | Server-defined URL for this resource.
`shortTarget`<br/>*string* | The name of the target resource.
`target`<br/>*string* |  The URL of the target resource to receive the matched traffic.

<!-------------------- CREATE A FORWARDING RULE -------------------->
##### Create a forwarding rule

```shell
curl -X POST \
  -H 'MC-Api-Key: your_api_key' \
  -H "Content-Type: application/json" \
  -d "request _body" \
  "https://cloudmc_endpoint/v1/services/gcp/test-area/forwardingrules"
```

<code>POST /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/forwardingrules</code>

Create a new forwarding rule

Required | &nbsp;
------- | -----------
`name`<br/>*string* | The display name of the forwarding rule

> `portRange`<br/>*string* | The destination ports that forwarding rule forwards packets with.

Reviewer: `ports` -> `port`, we don't handle a range as such on creation.

Reviewer: `The port which will be allocated to the forwarding rule. Supported ports are: 80 or 8080 (HTTP) and 443 (HTTPS)`

Reviewer: @JamieWulala I don't see any changes done here.

`shortTarget`<br/>*string* | The name of the target resource.

Optional | &nbsp;
------- | -----------
`description`<br/>*string* | Description of the forwarding rule.

> `reserveStatic`<br/>*boolean* | Whether the operation reserves a new static IP or not. The default value is false.

Reviewer: Can we rename this variable, it's not clear what it is, `reserveStaticIp`

Reviewer: need a bit more detail here, as the example JSON doesn't show all possible scenarios, which we could do.

If the value is false and if no `shortIPAddress` is provided, an ephemeral external IP address will be assigned. 
If the value is false and `shortIPAddress` is provided, then the existing external IP address will be assigned.
If the value is true, a new external static IP will be reserved and assigned.

Author: done

> `shortIPAddress`<br/>*string* | The name of a existing IP address.

Reviewer: we need a bit more details here.

The name of an existing global external IP to assign to the rule. This argument is only valid in conjunction with `reserveStaticIp` being `false`.

Author: done

Reviewer: @JamieWulala I don't see any changes done here.
